file-system.ts

```diff
@@ -151,15 +148,47 @@ export class FileSystem implements IFileSystem {
 	}
 
 	public readDirectory(path: string): IFuture<string[]> {
-		return this._readdir(path);
+		var future = new Future<string[]>();
+		fs.readdir(path, (err: Error, files: string[]) => {
+			if(err) {
+				future.throw(err);
+			} else {
+				future.return(files);
+			}
+		});
+		return future;
 	}
 
 	public readFile(filename: string): IFuture<NodeBuffer> {
-		return this._readFile(filename);
+		var future = new Future<NodeBuffer>();
+		fs.readFile(filename, (err: Error, data: NodeBuffer) => {
+			if(err) {
+				future.throw(err);
+			} else {
+				future.return(data);
+			}
+		});
+		return future;
 	}
 
-	public readText(filename: string, encoding?: string): IFuture<string> {
-		return <IFuture<string>> <any> this._readFile(filename, { encoding: encoding || "utf8" });
+	public readText(filename: string, options?: any): IFuture<string> {
+		options = options || { encoding: "utf8" };
+		if (_.isString(options)) {
```

Comment:
Can we simplify this? I suggest to introduce interface `IFsReadTextOptions: { encoding: string }`. This way this check will be removed.